includes/data-events/class-modal-checkout.php

```diff
@@ -0,0 +1,166 @@
+<?php
+/**
+ * Newspack Data Events Modal Checkout helper.
+ *
+ * @package Newspack
+ */
+
+namespace Newspack\Data_Events;
+
+use Newspack\Data_Events;
+use WP_Error;
+
+/**
+ * Class to register the Modal_Checkout listeners.
+ */
+final class Modal_Checkout {
+
+	/**
+	 * The name of the action for form submissions
+	 */
+	const FORM_SUBMISSION = 'form_submission_received';
+
+	/**
+	 * The name of the action for form submissions
+	 */
+	const FORM_SUBMISSION_SUCCESS = 'form_submission_success';
```

Comment:
Nitpick: The naming for this is a bit misleading. At first glance it seems to indicate checkout was successful, but it really indicates the modal checkout form has been opened (I think based on this comment below: `* Fires when a reader opens the modal checkout from a checkout button block.`).

I think we can stand to make these action names more clear. For example:

`MODAL_CHECKOUT_OPEN`
`MODAL_CHECKOUT_CONTINUE`
`MODAL_CHECKOUT_SUCCESS`

Reply:
I'll double check this one with @kmwilkerson! I agree it's a bit confusing given what's actually being tracked, but it's consistent with how stuff is being tracked (like as a successful form submission) and how it's captured otherwise in the plugin -- I'm not sure how closely we want to follow that though!